database/src/main/scala/cromwell/database/slick/SlickDatabase.scala

```diff
@@ -667,4 +668,61 @@ class SlickDatabase(databaseConfig: Config) extends SqlDatabase {
 
     runTransaction(action)
   }
+
+  override def addMetadataEvent(workflowUuid: String,
+                                key: String,
+                                value: String,
+                                timestamp: Timestamp)(implicit ec: ExecutionContext): Future[Unit] = {
+
+    val action = DBIO.seq(dataAccess.metadataAutoInc += Metadatum(workflowUuid, key, callFqn = None, index = None, attempt = None, Option(value), timestamp))
```

Comment:
`DBIO.seq` is an abomination, that reads to me as "[just run already!](http://slick.lightbend.com/doc/3.1.1/api/index.html#slick.dbio.DBIOAction$@seq[E<:slick.dbio.Effect](actions:slick.dbio.DBIOAction[_,slick.dbio.NoStream,E]*):slick.dbio.DBIOAction[Unit,slick.dbio.NoStream,E])" It consumes practically anything, and then returns `DBIO[Unit]`.<sup>1</sup>

When I've used it in the past, it was because I couldn't get anything else to work for a while. I've since associated it with code smell. I'm not sure if you ran into a similar issues. Perhaps the method return was argumentative, as you wanted to throw away the result of the insert and return `Unit`.

If that's the case, try mapping the result outside the transaction via:

```scala
val action = dataAccess.metadataAutoInc += Metadatum(workflowUuid, key,
  callFqn = None, index = None, attempt = None, Option(value), timestamp)
runTransaction(action).map(_ => ())
```

or map to a `Unit` using a for comprehension (easier to read IMHO, but `map`s inside the transaction):

```scala
val action = for {
  _ <- dataAccess.metadataAutoInc += Metadatum(workflowUuid, key,
    callFqn = None, index = None, attempt = None, Option(value), timestamp)
} yield ()
runTransaction(action)
```

---
<sup>1</sup> FYI, for _actually_ running multiple operations, `DBIO.seq`'s sibling [`DBIO.sequence`](http://slick.lightbend.com/doc/3.1.1/api/index.html#slick.dbio.DBIOAction$@sequence[R,M[+_]<:TraversableOnce[_],E<:slick.dbio.Effect](in:M[slick.dbio.DBIOAction[R,slick.dbio.NoStream,E]])(implicitcbf:scala.collection.generic.CanBuildFrom[M[slick.dbio.DBIOAction[R,slick.dbio.NoStream,E]],R,M[R]]):slick.dbio.DBIOAction[M[R],slick.dbio.NoStream,E]) is cool with me, as it behaves more like [`Future.sequence`](http://www.scala-lang.org/api/2.11.8/index.html#scala.concurrent.Future$@sequence[A,M[X]<:TraversableOnce[X]](in:M[scala.concurrent.Future[A]])(implicitcbf:scala.collection.generic.CanBuildFrom[M[scala.concurrent.Future[A]],A,M[A]],implicitexecutor:scala.concurrent.ExecutionContext):scala.concurrent.Future[M[A]]) and doesn't erase types by returning a wrapped `Unit`.